src/approaches/nsrt_rl_approach.py

```diff
@@ -72,10 +90,128 @@ def learn_from_interaction_results(
+
+                if curr_option.terminal(s):
+                    # TODO: make this not hardcoded to be environment specific,
+                    # or implement specifically per environment and throw an
+                    # error for environments where this is not implemented.
+
+                    # Check if we reached our subgoal within a tolerance by
+                    # checking the difference between our proposed subgoal
+                    # (which is expressed relatively) and the relative changes
+                    # that actually happened.
+                    if curr_option.params[-1] > 0:  # if holding becomes true
+                        dblock = self.infer_delta(block, curr_states, ['grasp'])
+                        drobot = self.infer_delta(robot, curr_states, ['x', 'y', 'grip', 'holding'])
+                    else:
+                        dblock = self.infer_delta(block, curr_states, ['x', 'grasp'])
+                        drobot = self.infer_delta(robot, curr_states, ['x', 'grip', 'holding'])
+                    actual_delta = np.array(dblock + drobot)
+                    if np.allclose(curr_option.params, actual_delta, atol=self._reward_epsilon):
+                        reward = self._pos_reward
+                    else:
+                        reward = self._neg_reward
+                    curr_rewards.append(reward)
+
+                    # Store transition data.
+                    option_to_data[curr_option.name].append([curr_states, curr_actions, curr_rewards, curr_relative_params])
+
+                    # Advance to next option.
+                    curr_option_idx += 1
+                    if curr_option_idx < len(plan):
+                        curr_option = plan[curr_option_idx]
+                        if curr_option.name not in option_to_data:
+                            option_to_data[curr_option.name] = []
+                    else:
+                        # If we run out of options in the plan, there should be
+                        # an _OptionPlanExhausted exception, and so there is
+                        # nothing more in the trajectory that we have not yet
+                        # assigned to an _Option already.
+                        # TODO: maybe add an assert to confirm the above ^
+                        pass
+
+                    # Initialize trajectory for next option.
+                    curr_states = [s]
+                    curr_actions = []
+                    curr_rewards = []
+                    curr_relative_params = []
+
+                else:  # case where terminal state not reached.
+                    curr_rewards.append(self._neg_reward)
+                    # Handle the case where we are at the last state in the
+                    # trajectory, but it is not a terminal state of the current
+                    # option. This occurs when the plan we are executing is a
+                    # partial refinement.
```

Comment:
I could also imagine this happening if we just run out of horizon in the environment. maybe we should just ditch this data because we can't know if the option would have succeeded with just a few more steps?

Reply:
To be clear, the limited horizon is coming from `for _ in range(max_num_steps):` in `utils.run_policy()`. 

This is an interesting point: 
It is true that perhaps this option would've succeeded in more steps. By penalizing it negatively in such a case, maybe it will learn that what it was doing was "wrong". The implicit assumption here is that `max_num_steps` encodes our belief for the sufficient number of steps to solve a task. As long as we take `max_num_steps` to be sufficiently high (higher than our oracle options take), then it's not a problem? 

If `max_num_steps` was set high enough, ditching this data is bad, because the option will never get a signal that what it was doing here was bad. We may see the option continuing to perform this behavior, and we'll continue to not assign it a positive or negative reward (since we ditch it).

Reply:
Note from discussion: 
Maybe dense rewards addresses this issue.